gateway/src/http_entrypoints.rs

```diff
@@ -845,10 +846,31 @@ impl GatewayApi for GatewayImpl {
         apictx.latencies.instrument_dropshot_handler(&rqctx, handler).await
     }
 
-    async fn ignition_get(
+    async fn ignition_list_v2(
+        rqctx: RequestContext<Self::Context>,
+    ) -> Result<HttpResponseOk<Vec<ignition::v2::SpIgnitionInfo>>, HttpError>
+    {
+        let apictx = rqctx.context();
+        let mgmt_switch = &apictx.mgmt_switch;
+        let handler = async {
+            let out = mgmt_switch
+                .bulk_ignition_state()
+                .await?
+                .map(|(id, state)| ignition::v2::SpIgnitionInfo {
+                    id: id.into(),
+                    details: state.into(),
+                })
+                .collect();
+
+            Ok(HttpResponseOk(out))
+        };
```

Comment:
I'm assuming that we'll never have enough items to warrant pagination here?

Reply:
Interesting question! At least in MGS it does look to possibly allow it https://github.com/oxidecomputer/management-gateway-service/blob/91c9d622027c39f5acdd17ae3dab046330bf6717/gateway-messages/src/sp_impl.rs#L765-L776 but I'm guessing the amount of data returned right now is small enough we just haven't bothered